src/wazuh_db/wdb_global.c

```diff
@@ -1058,17 +1058,45 @@ wdbc_result wdb_global_sync_agent_info_get(wdb_t *wdb, int* last_agent_id, char
 }
 
 char* wdb_global_calculate_agent_group_csv(wdb_t *wdb, int id) {
-    cJSON* j_agent_groups = wdb_global_select_group_belong(wdb, id);
     char* result = NULL;
-    if (j_agent_groups) {
-        cJSON* j_group_name = NULL;
-        cJSON_ArrayForEach(j_group_name, j_agent_groups) {
-            wm_strcat(&result, cJSON_GetStringValue(j_group_name), MULTIGROUP_SEPARATOR);
+    sqlite3_stmt *stmt = NULL;
+
+    if (!wdb->transaction && wdb_begin2(wdb) < 0) {
+        mdebug1("Cannot begin transaction");
+        return NULL;
+    }
+
+    if (wdb_stmt_cache(wdb, WDB_STMT_GLOBAL_SELECT_GROUP_BELONG) < 0) {
+        mdebug1("Cannot cache statement");
+        return NULL;
+    }
+
+    stmt = wdb->stmt[WDB_STMT_GLOBAL_SELECT_GROUP_BELONG];
+
+    if (sqlite3_bind_int(stmt, 1, id) != SQLITE_OK) {
+        merror("DB(%s) sqlite3_bind_int(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+        return NULL;
+    }
+
+    int _status = SQLITE_ROW;
+
+    while ((_status = wdb_step(stmt)) == SQLITE_ROW) {
+        char * group_hash = (char *) sqlite3_column_text(stmt, 0);
+
+        if (group_hash == NULL) {
+            mdebug1("Group hash is NULL");
+            continue;
         }
-        cJSON_Delete(j_agent_groups);
+
+        if (result != NULL && WDB_MAX_RESPONSE_SIZE < strlen(result) + strlen(group_hash) + 1) {
+            mdebug1("The agent's groups exceed the socket maximum response size.");
+            break;
+        }
+        wm_strcat(&result, group_hash, MULTIGROUP_SEPARATOR);
     }
-    else {
-        mdebug1("Unable to get groups of agent '%03d'", id);
+
+    if (SQLITE_DONE != _status) {
+        mdebug1("SQL statement execution failed");
```

Comment:
Same here, in case of error we must fail instead of return an incomplete string.

Reply:
Yeap, I think several checks are missing even before this implementation, this wdb code needs a deeper revision.

Reply:
I agree.
We'll leave like it is because we don't have time.